src/core/geometry/qgssfcgalgeometry.h

```diff
@@ -0,0 +1,516 @@
+/***************************************************************************
+                         qgssfcgalGeometry.h
+                         -------------------
+    begin                : May 2025
+    copyright            : (C) 2025 by Oslandia
+    email                : benoit dot de dot mezzo at oslandia dot com
+    email                : jean dot felder at oslandia dot com
+    email                : loic dot bartoletti at oslandia dot com
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#ifdef WITH_SFCGAL
+#ifndef QGSSGCGAL_GEOMETRY_H
+#define QGSSGCGAL_GEOMETRY_H
+
+#define SIP_NO_FILE
+
+#include "qgis_core.h"
+#include "qgis_sip.h"
+#include "qgsabstractgeometry.h"
+#include "qgscircularstring.h"
+#include "qgscompoundcurve.h"
+#include "qgscurvepolygon.h"
+#include "qgspoint.h"
+#include "qgspolygon.h"
+#include "qgslinestring.h"
+#include "qgsmulticurve.h"
+#include "qgsmultilinestring.h"
+#include "qgsmultipoint.h"
+#include "qgsmultipolygon.h"
+#include "qgsmultisurface.h"
+#include "qgspolyhedralsurface.h"
+#include "qgstriangulatedsurface.h"
+#include "qgstriangle.h"
+#include "qgssfcgalengine.h"
+#include "qgslogger.h"
+
+/**
+ * Wraps SFCGAL geometry object and its QgsAbstractGeometry source
+ * \ingroup core
+ * \class QgsSfcgalGeometry
+ * \brief SfcgalGeometry geometry type.
+ * \since QGIS 3.44
+ */
+class CORE_EXPORT QgsSfcgalGeometry
+{
+  public:
+    //! Constructor for an empty sfcgalGeometry geometry.
+    QgsSfcgalGeometry();
+
+    //! Constructor with SFCGAL shared ptr
+    QgsSfcgalGeometry( sfcgal::shared_geom sfcgalGeom );
+
+    //! Constructor with QgsAbstractGeometry unique_ptr
+    QgsSfcgalGeometry( std::unique_ptr<QgsAbstractGeometry> &qgsGeom );
+
+    //! Constructor with QgsAbstractGeometry ptr
+    QgsSfcgalGeometry( const QgsAbstractGeometry &qgsGeom );
+
+    //! Constructor with QgsAbstractGeometry ptr
+    QgsSfcgalGeometry( const QgsGeometry &qgsGeom );
```

Comment:
Can you add documentation on the lifetime / ownership requirements of the qgis geometries? It's not immediately clear from the API (and the ownership here differs from the geos equivalent)